### PR TITLE
[FEATURE][SWE-1552][override useCollection and booking show componets]

### PR DESCRIPTION
**Pull request by aleksandrgolyk**

### CODE REVIEW

File: src/__constants__/statuses.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const STATUSES = {
+  BOOKED: 'BOOKED',
+  PENDING: 'PENDING',
+  CANCELED: 'CANCELED',
+  FUTURE: 'Future',
```
Comment (alexpassss) [inline src/__constants__/statuses.js]: why text formatting is different?

File: src/domains/Booking/components/BookingList/BookingList.js [hunk 2]
```diff
@@ -1,26 +1,33 @@
...
 import { useHistory } from 'react-router-dom'
+import { useTranslations } from 'contexts/Translation'
 
 const BookingList = (props) => {
-  const { bookings, title } = props
+  const { bookings, title, bookingCounter, loadMoreAvailable, next } = props
   const history = useHistory()
+  const { t } = useTranslations()
+
   const [selectedItem, setSelectedItem] = useState(null)
```
Comment (alexpassss) [inline src/domains/Booking/components/BookingList/BookingList.js]: refactor this part. create separate hooks.
ex: for selection feature

File: src/domains/Booking/components/BookingList/BookingList.js [hunk 3]
```diff
@@ -1,26 +1,33 @@
...
   const onCreateButtonClick = () => history.push('/booking/create')
   const onEmptySpaceClick = () => setSelectedItem(null)
+  // [COMPUTED PROPERTIES]
+
   const checkRenderConditions = props?.listView ? 12 : [12, 12, 6, 6, 6, 4]
   const isCollapsible = !bookings?.length && 'disabled'
   const defaultPanel = isPanelOpen && title
 
+  const bgColor =
+    bookings?.length > 0 ? 'var(--badge-color)' : 'var(--btn-default-bg)'
```
Comment (alexpassss) [inline src/domains/Booking/components/BookingList/BookingList.js]: --btn-default-bg for badge? Is it correct?

File: src/domains/Booking/components/BookingsByType/BookingsByType.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  const {
+    additionalQuery,
+    bookingsRef,
+    bookingTitle,
+    bookingsCounterName,
+    bookingsBelongTo
+  } = props
+
+  const { t } = useTranslations()
+  const computetBookingsRef = useMemo(() => {
```
Comment (alexpassss) [inline src/domains/Booking/components/BookingsByType/BookingsByType.js]: this part should be incapsulated inside useBookings hook

same with loadMoreAvailable variable below

File: src/domains/Booking/components/BookingsByType/BookingsByType.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+import { useBookings, useGetBookingsCounter } from '../../../Booking/hooks/get'
+
+import BookingList from '../BookingList'
+import PropTypes from 'prop-types'
+import { useTranslations } from 'contexts/Translation'
+
+const { BOOKINGS } = COLLECTIONS
+function BookingsByType(props) {
+  const {
+    additionalQuery,
```
Comment (alexpassss) [inline src/domains/Booking/components/BookingsByType/BookingsByType.js]: please add comments here

File: src/domains/Booking/hooks/get/useBookings/useBookings.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import useCollection from 'hooks/useCollection'
+
+const ref = 'bookings'
```
Comment (alexpassss) [inline src/domains/Booking/hooks/get/useBookings/useBookings.js]: duplicated code

File: src/domains/Booking/hooks/get/useBookings/useGetBookings.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+import firebase from 'firebase/compat/app'
+import moment from 'moment'
+import { useCollectionData } from 'react-firebase-hooks/firestore'
+
+const { CURRENT_DATE_FORMAT } = TIME
+const { BOOKED, CANCELED } = STATUSES
+const { BOOKINGS } = COLLECTIONS
+
+const useGetBookings = ({ additionalQuery }) => {
```
Comment (alexpassss) [inline src/domains/Booking/hooks/get/useBookings/useGetBookings.js]: please add comments here

File: src/hooks/useCollection.js [hunk 8]
```diff
@@ -91,3 +108,97 @@ const useCollection = (props) => {
 }
 
 export default useCollection
+
+// import {
```
Comment (alexpassss) [inline src/hooks/useCollection.js]: please remove comments

File: src/pages/App/Booking/BookingsAll.js [hunk 9]
```diff
@@ -1,32 +1,62 @@
...
-    clientPassedBookings,
-    passedBookingsLoading,
-    passedBookingError,
-    clientCanceledBookings,
-    canceledBookingsLoading,
-    canceledBookingError
-  ] = useGetBookings()
+  const { user } = useUser()
+  const currentDateFormatted = moment().format(CURRENT_DATE_FORMAT)
+  // const [
```
Comment (alexpassss) [inline src/pages/App/Booking/BookingsAll.js]: useless comments

File: src/pages/App/Booking/BookingsAll.js [hunk 10]
```diff
@@ -59,7 +110,7 @@ const BookingsAll = () => {
             bookings={clientCanceledBookings}
           />
         ) : null}
-        {computedError ? <Text>{JSON.stringify(computedError)}</Text> : null}
+        {computedError ? <Text>{JSON.stringify(computedError)}</Text> : null} */}
```
Comment (alexpassss) [inline src/pages/App/Booking/BookingsAll.js]: useless comment

File: src/pages/App/Booking/BookingsAll.js [hunk 11]
```diff
@@ -37,7 +67,28 @@ const BookingsAll = () => {
...
+          bookingsCounterName="passedBookingsCounter"
+        />
+        <BookingsByType
+          bookingsBelongTo={user}
+          additionalQuery={additionalQuery}
+          bookingsRef={canceledBookingsRef}
+          bookingTitle={canceledTitle}
+          bookingsCounterName="canceledBookingsCounter"
+        />
+        {/* {loading ? <Text>Loading bookings...</Text> : null}
```
Comment (alexpassss) [inline src/pages/App/Booking/BookingsAll.js]: useless comment

**Review by alexpassss — CHANGES_REQUESTED** [review]

File: src/__constants__/statuses.js (hunk 1 above)
Comment (aleksandrgolyk) [reply src/__constants__/statuses.js]: fixed

Comment (Arthur-Pochinok-MENTA) [inline src/__constants__/statuses.js]: Why `CANCELED` status equals to `'CANCELED'`, but `FUTURE` - to `'Future'`?! Use same style (uppercase)

File: src/domains/Booking/components/BookingList/BookingList.js [hunk 12]
```diff
@@ -1,26 +1,33 @@
...
+  const { t } = useTranslations()
+
   const [selectedItem, setSelectedItem] = useState(null)
   const isPanelOpen = useMemo(
     () => JSON.parse(localStorage.getItem(title)),
     [title]
   )
   const onCreateButtonClick = () => history.push('/booking/create')
   const onEmptySpaceClick = () => setSelectedItem(null)
+  // [COMPUTED PROPERTIES]
```
Comment (Arthur-Pochinok-MENTA) [inline src/domains/Booking/components/BookingList/BookingList.js:21]: Redundant line

File: src/domains/Booking/components/BookingsByType/BookingsByType.js [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+    bookingsRef,
+    bookingTitle,
+    bookingsCounterName,
+    bookingsBelongTo
+  } = props
+
+  const { t } = useTranslations()
+  const computetBookingsRef = useMemo(() => {
```
Comment (Arthur-Pochinok-MENTA) [inline src/domains/Booking/components/BookingsByType/BookingsByType.js]: `computeD...`

File: src/domains/Booking/components/BookingsByType/BookingsByType.js [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+  )
+}
+
+BookingsByType.propTypes = {
+  bookingsBelongTo: PropTypes.object.isRequired,
+  bookingsRef: PropTypes.array.isRequired,
+  bookingTitle: PropTypes.string.isRequired,
+  additionalQuery: PropTypes.array.isRequired,
+  bookingsCounterName: PropTypes.oneOf([
+    'futureBookingsCounter',
```
Comment (Arthur-Pochinok-MENTA) [inline src/domains/Booking/components/BookingsByType/BookingsByType.js]: Maybe, it makes sense to move this to the constant..

File: src/domains/Booking/hooks/get/useBookings.js [hunk 15]
```diff
@@ -0,0 +1,9 @@
+import useCollection from 'hooks/useCollection'
+
+const ref = 'bookings'
```
Comment (Arthur-Pochinok-MENTA) [inline src/domains/Booking/hooks/get/useBookings.js]: It's better to move this variable to app constants...

File: src/domains/Booking/hooks/get/useBookings/useBookings.js [hunk 16]
```diff
@@ -0,0 +1,9 @@
...
+export default function useBookings(props) {
```
Comment (Arthur-Pochinok-MENTA) [inline src/domains/Booking/hooks/get/useBookings/useBookings.js]: This file is identical to `src/domains/Booking/hooks/get/useBookings.js`

File: src/hooks/useCollection.js [hunk 17]
```diff
@@ -15,48 +15,62 @@ import firestore from 'services/firebase/firestore'
...
+          const documents = data?.docs?.map((item) => item?.data())
+          setValue(documents)
+          if (propsLimit) {
+            setLastVisible(data?.docs?.[data?.docs?.length - 1])
+          }
           setLoading(false)
         })
       } catch (err) {
+        // eslint-disable-next-line no-console
+        console.log(err)
```
Comment (Arthur-Pochinok-MENTA) [inline src/hooks/useCollection.js]: There is special type of console output function for the errors, - `console.error()`...

File: src/hooks/useCollection.js [hunk 18]
```diff
@@ -15,48 +15,62 @@ import firestore from 'services/firebase/firestore'
...
+                ? orderBy(...propsOrderBy)
+                : orderBy(...baseSortRule),
+              limit(propsLimit)
             )
           : query(
-              collection(firestore, props?.ref),
-              Array.isArray(props?.orderBy)
-                ? orderBy(...props?.orderBy)
+              collection(firestore, propsRef),
+              Array.isArray(propsOrderBy)
```
Comment (Arthur-Pochinok-MENTA) [inline src/hooks/useCollection.js]: This and 2 following rows are equal to rows 45-47, please figure out on it due to DRY principle

File: src/pages/App/Booking/BookingsAll.js [hunk 19]
```diff
@@ -1,32 +1,46 @@
...
-    canceledBookingError
-  ] = useGetBookings()
   const goToNextBookingPage = () => {
     history.push('/next-booking')
   }
-  const computedError =
-    futureBookingError || canceledBookingError || passedBookingError
+
+  const futureBookingsRef = [
+    ['userId', '==', user?._id],
```
Comment (Arthur-Pochinok-MENTA) [inline src/pages/App/Booking/BookingsAll.js:29]: Don't u think that this row is an exact value of `additionalQuery` variable ?🙂
Figure out somehow rows below as well

**Review by Arthur-Pochinok-MENTA — CHANGES_REQUESTED** [review]

File: src/__constants__/statuses.js (hunk 1 above)
Comment (aleksandrgolyk) [reply src/__constants__/statuses.js]: fixed

File: src/domains/Booking/components/BookingsByType/BookingsByType.js (hunk 4 above)
Comment (aleksandrgolyk) [reply src/domains/Booking/components/BookingsByType/BookingsByType.js]: done

**Review by alexpassss — APPROVED** [review]

**Review by GingerBoy14 — APPROVED** [review]

**Pull request merged**